### DISCUSSION TITLE
How to connect to localhost / foundry / anvil?

### DISCUSSION THREAD

**Discussion by ian** [rainbow-me/rainbowkit#1109] [Q&A] [answered]
Not sure how to connect to a local-running Anvil connection, nothing in the docs.

All of the following do NOT work:

```
import { foundry } from "wagmi/chains"
const { chains, provider, webSocketProvider } = configureChains(
  [foundry],
  [jsonRpcProvider({ rpc: () => ({ http: "http://localhost:8545" }) })]
);
```

```
import { localhost } from "wagmi/chains"
const { chains, provider, webSocketProvider } = configureChains(
  [localhost],
  [jsonRpcProvider({ rpc: () => ({ http: "http://localhost:8545" }) })]
);
```

The result is always a generic list of eth/goerli chains:

<img width="461" alt="CleanShot 2023-03-23 at 16 13 26@2x" src="https://user-images.githubusercontent.com/19853/227342599-8f5092c0-fcc2-4486-9e84-1f6529fab283.png">

**Comment by zkovac58** [comment, marked as answer]
In the latest version (v2), Anvil can be imported from 'wagami/chains' ...

`import {
  mainnet,
  polygon,
  optimism,
  arbitrum,
  base,
  anvil,
} from 'wagmi/chains';`


and then added to the chains array...

`const config = getDefaultConfig({
  appName: 'myapp',
  projectId: 'projectId',
  chains: [mainnet, polygon, optimism, arbitrum, base, anvil],
  ssr: false,
});
`

This worked for me.
